src/admin/admin.controller.ts

```diff
@@ -1,4 +1,166 @@
-import { Controller } from '@nestjs/common';
+import { Body, Controller, Get, HttpStatus, Param, ParseIntPipe, Post } from '@nestjs/common';
+import { AdminService } from './admin.service';
+import { TransactionService } from 'src/transactions/transactions.service';
+import { UserService } from 'src/user/user.service';
+import { TransactionType } from '@prisma/client';
+import { ProviderService } from 'src/provider/provider.service';
+import { CreditsDto } from './dto/credits.dto';
+import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { Transaction } from 'src/transactions/dto/transactions.dto';
+import { WalletCredits } from 'src/wallet/dto/wallet.dto';
 
+
+@ApiTags('admin')
 @Controller('admin')
-export class AdminController {}
+export class AdminController {
+    constructor(
+        private transactionService: TransactionService,
+        private userService: UserService,
+        private adminService: AdminService,
+        private providerService: ProviderService
+    ) {}
+
+    @ApiOperation({ summary: 'Get All Users Transactions' })
+    @ApiResponse({ status: HttpStatus.OK, description: 'transactions fetched successfully', type: [Transaction] })
+    @Get("/:adminId/transactions/users")
+    // get all transactions of all users
+    async getAllUsersTransactions(
+        @Param("adminId", ParseIntPipe) adminId: number
+    ) {
+        // check admin
+        await this.adminService.getAdminWallet(adminId);
+        
+        // fetch transactions
+        const transactions = await this.transactionService.fetchAllUsersTransactions();
+        return {
```

Comment:
Use nestjs builtin @res for responses in all APIs, instead of manually creating JSON object.